otherlibs/fiber/pool.ml

```diff
@@ -1,17 +1,13 @@
 open Stdune
 open Core
-open Core.O
-
-type mvar =
-  | Done
-  | Task of (unit -> unit t)
 
 type status =
   | Open
   | Closed
 
-type t =
-  { mvar : mvar Mvar.t
+type nonrec t =
+  { writers : (unit -> unit t) Queue.t
```

Comment:
These are tasks, not writers, right?

```suggestion
  { tasks : (unit -> unit t) Queue.t
```

Reply:
Marked as resolved but still says "writers"?

Reply:
I've made the change locally but I'm waiting for the tests PR to be merged as this PR also now includes fixes to the newly added tests.

Reply:
Pushed.